src/server/events/index.ts

```diff
@@ -0,0 +1,102 @@
+import ical from 'node-ical';
+import { AuthenticationError } from 'apollo-server-express';
+import { ObjectID } from 'mongodb';
+import { EventUpdateInput, EventDbObject } from '../generated/graphql';
+import { Models } from '../models';
+import { EventUpdate } from '../../client/routes/events/ManageEventTypes';
+
+const filterByCalType = (objNames: string[], cal: Record<string, any>, type: string): string[] => {
+	return objNames.filter(obj => cal[obj].type === type);
+};
+
+const filterCalByObjectNames = (
+	objNames: string[],
+	cal: Record<string, any>
+): Record<string, any> => {
+	return objNames.map(key => cal[key]);
+};
+
+export const transformCalEventToDBUpdate = (event: Record<string, any>): EventUpdate => {
+	if (!event.start || !event.end)
+		throw new AuthenticationError('Calendar event did not contain start or end timestamp');
+	const parsedStart = new Date(event.start);
+	const parsedEnd = new Date(event.end);
+	const parsedType = /\[(.*?)\]/.exec(event.summary); // Looks for a **single** [Event Type] tag in name
+	return {
+		name: event.summary.replace(/\s*\[(.*?)\]\s*/, ''),
+		startTimestamp: event.start,
+		duration: Math.floor((parsedEnd.getTime() - parsedStart.getTime()) / (1000 * 60)),
+		description: event.description,
+		location: event.location,
+		eventType: parsedType != null ? parsedType[1] : '',
+	} as EventUpdate;
+};
+
+export async function pullCalendar(calendarID: string | undefined): Promise<EventUpdate[] | null> {
```

Comment:
i think if calendarId is undefined or null, should throw error

Reply:
Done

Reply:
Done